### PR TITLE
CONV-1435: Gravity layout

### PR DESCRIPTION
**Pull request by gilgtm**
- Adds a new Chat App demo and a new behavior called verticalLayoutGravity. When verticalLayoutGravity is set to bottom, scrolling works the way you would expect for a messaging app.

https://github.com/square/Listable/assets/73715873/a11740aa-8432-492f-93ba-50fbd0540422

### Checklist

Please do the following before merging:

- [x] Ensure any public-facing changes are reflected in the [changelog](https://github.com/kyleve/Listable/blob/main/CHANGELOG.md). Include them in the `Main` section.

### CODE REVIEW

**Review by nononoah — APPROVED** [review]
Previously reviewed in nearly this state and approved. It would be good to get someone else from UI Systems to weigh in, or even Kyle.

File: ListableUI/Sources/ListView/ListView.swift [hunk 1]
```diff
@@ -1617,8 +1620,59 @@ fileprivate extension UIScrollView
...
+                guard isContentScrollable else { return }
+                
+                let heightDelta = contentSize.height - oldValue.height
+                guard heightDelta > 0 else { return }
+                
+                let maxContentOffsetY = contentSize.height - bounds.height + adjustedContentInset.bottom
+                let targetY = self.contentOffset.y + heightDelta
+
+                self.contentOffset.y = min(targetY, maxContentOffsetY)
+            }
```
Comment (kyleve) [inline ListableUI/Sources/ListView/ListView.swift:1644]: Can we add an `assert` if `.bottom` is provided, but the layout direction is `.horizontal`? Just so folks know it's not supported.

**Review by kyleve — APPROVED** [review]
Really nice work – thank you for getting this across the line!

File: ListableUI/Sources/ListView/ListView.swift (hunk 1 above)
Comment (gilgtm) [reply ListableUI/Sources/ListView/ListView.swift:1644]: - [ ] todo

**Pull request merged**